### PR TITLE
React accordion component

### PR DESCRIPTION
**Pull request by shawnkost**
react-accordion

### CODE REVIEW

File: react-accordion-component/src/accordion.jsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    return (
+      <div className="container">
+        <Topics topics={this.props.topics} test={this.test} />
+      </div>
+    );
+  }
+}
+
+function TopicDetails(props) {
+  return <div className="hide">{props.value.details}</div>
```
Comment (scbowler) [inline react-accordion-component/src/accordion.jsx:34]: Here you can compare the `id` stored in state with the `id` of the item, if the ids match that is the open item and add a class of `show`, if the ids do not match add a class of `hide`

File: react-accordion-component/src/accordion.jsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+function TopicName(props) {
+  return <div className="topics" onClick={props.test}>{props.value.name}</div>
+}
+
+function Topics(props) {
+  const topics = props.topics;
+  const newTopics = topics.map((topic) => {
+    return (
+      <div>
+        <TopicName key={topic.name} value={topic} test={props.test} />
```
Comment (scbowler) [inline react-accordion-component/src/accordion.jsx:46]: Here you can save the correct id with the click handler, In the below example I will use your names but you need to change the prop name from `test` to something more appropriate. Also make sure to use something unique for the `key` prop like an `id`, `name` is not guaranteed to be unique

```JSX
<TopicName key={topic.id} value={topic} test={() => props.test(topic.id)} />
```

File: react-accordion-component/src/index.jsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  {
+    name: "Cascading Style Sheets",
+    details:
+      "CSS is a style sheet language used for describing the presentation of a document written in a markup language such as HTML.[1] CSS is a cornerstone technology of the World Wide Web, alongside HTML and JavaScript.[2]",
+  },
+  {
+    name: "JavaScript",
+    details:
+      "JavaScript often abbreviated as JS, is a programming language that conforms to the ECMAScript specification.[7] JavaScript is high-level, often just-in-time compiled, and multi-paradigm. It has curly-bracket syntax, dynamic typing, prototype-based object-orientation, and first-class functions.",
+  },
```
Comment (scbowler) [inline react-accordion-component/src/index.jsx:20]: Each item needs an `id` property that is unique

**Review by scbowler — CHANGES_REQUESTED** [review]
Some updates needed, let us know if you have any questions

File: react-accordion-component/src/accordion.jsx (hunk 2 above)
Comment (shawnkost) [reply react-accordion-component/src/accordion.jsx:46]: I'm confused on what the included snippet is doing, and how I can access this within handleClick(). I have attached a photo of my updated code. 

`test={() => props.test(topic.id)}`

updated-code

Comment (scbowler) [reply react-accordion-component/src/accordion.jsx:46]: In `handleClick` `event` should now be the `id` of the item clicked on. Try just console logging event without the `target` property within `handleClick` it should be the id. You'll want to update the name of the parameter to reflect that. Then you just need to save the `id` into state.

**Review by scbowler — APPROVED** [review]
Much better! Looks great! 👍 

FYI: you can accomplish the same thing with just one property in state.... 😄

**Pull request merged**